File: den-3431/parity/verify.rs
use std::{env, fs};
fn main() {
    let text = fs::read_to_string(env::args().nth(1).expect("path")).expect("read");
    assert!(text.contains("\"schema\":\"file-tunnel-test/object-storage-evidence/v2\""));
    assert!(text.contains("\"tamperDetected\":true"));
    assert!(text.contains("\"rangeVerified\":true"));
    assert!(text.contains("\"staleCapabilityRejected\":true"));
    assert!(text.contains("\"expiredCapabilityRejected\":true"));
    assert!(text.contains("\"wrongEtagRejected\":true"));
    assert!(text.contains("\"serverModeMetadataVerified\":true"));
    assert!(text.contains("\"crossPrefixReadRejected\":true"));
    assert!(text.contains("\"partialResponses\":1"));
    assert!(text.contains("\"finalDeletionEvidence\":\"verified\""));
    assert!(text.contains("\"tenantScopedCleanup\":true"));
    assert!(!text.contains("BEGIN PRIVATE KEY"));
    println!("rust parity: PASS");
}

File: den-3431/parity/Verify.java
import java.nio.file.*;
public final class Verify {
  private static void require(boolean value, String message) { if (!value) throw new AssertionError(message); }
  public static void main(String[] args) throws Exception {
    String text = Files.readString(Path.of(args[0]));
    require(text.contains("\"schema\":\"file-tunnel-test/object-storage-evidence/v2\""), "schema");
    require(text.contains("\"tamperDetected\":true"), "tamper");
    require(text.contains("\"staleCapabilityRejected\":true"), "stale capability");
    require(text.contains("\"reorderedPartRejected\":true"), "reordered part");
    require(text.contains("\"expiredCapabilityRejected\":true"), "expired capability");
    require(text.contains("\"wrongEtagRejected\":true"), "wrong ETag");
    require(text.contains("\"partialResponses\":1"), "partial response fault");
    require(text.contains("\"finalDeletionEvidence\":\"verified\""), "deletion");
    require(text.contains("\"tenantScopedCleanup\":true"), "tenant cleanup");
    require(!text.contains("X-Amz-Signature"), "signed URL leaked");
    System.out.println("java parity: PASS");
  }
}
